### PR TITLE
Koatuu_service

### PR DESCRIPTION
**Pull request by Anntroy**
This version fills all the koatuu fields for all current records.

### CODE REVIEW

File: data_ocean/services/koatuu_service.py [hunk 1]
```diff
@@ -15,33 +15,156 @@ class KoatuuConverter(Converter):
...
+                koatuu_dict[unit_name]=table_record
+        return koatuu_dict
+    
+
+    #creating list out of name and district_id from json file for district items
+    def create_district_items_list(self, object_koatuu, object_region_name, region_dict, object_district_name):
+        district_items_list = []
+        if not (object_region_name in region_dict):
+            return
+        else:
```
Comment (criticus) [inline data_ocean/services/koatuu_service.py:68]: this `else` is not needed

File: data_ocean/services/koatuu_service.py [hunk 2]
```diff
@@ -15,33 +15,156 @@ class KoatuuConverter(Converter):
...
+            region_koatuu = region_dict[object_region_name]            
+            if not (str(object_koatuu[self.LEVEL_TWO])[:2] == str(region_koatuu.koatuu)[:2]):
+                return
+            else:
```
Comment (criticus) [inline data_ocean/services/koatuu_service.py:72]: this one too

File: data_ocean/services/koatuu_service.py [hunk 3]
```diff
@@ -15,33 +15,156 @@ class KoatuuConverter(Converter):
...
+                object_district_name = object_district_name + str(region_koatuu.id)
+                district_items_list.append(object_district_name)
+        return district_items_list
+
+    #creating list out of name and district_id from json file fot city items
+    def create_city_items_list(self, data, object_koatuu, city_dict, district_items_list):
+        city_list =[]
+        for object_district_name in district_items_list:
+            if (object_district_name in city_dict) & (str(object_koatuu[self.LEVEL_TWO])[2]=='1'):
```
Comment (criticus) [inline data_ocean/services/koatuu_service.py:81]: what does `1` mean? please use Const to make it more clear

File: data_ocean/services/koatuu_service.py [hunk 4]
```diff
@@ -15,33 +15,156 @@ class KoatuuConverter(Converter):
...
+                object_level_four = object_koatuu[self.LEVEL_FOUR]
+                self.save_to_city_table(data, object_koatuu, object_level_four, district_dict, city_dict, object_citydistrict_name, district_items_list)
+                self.save_to_citydistrict_table(data, object_koatuu, object_level_four, citydistrict_dict, object_citydistrict_name, city_items_list, city_dict)
         print("Koatuu values saved")
+       
+    # #writing entry to koatuu field in region table
+    def save_to_region_table(self, data, object_koatuu, region_dict, object_region_name):
+        if  not object_region_name in region_dict:
+            return
+        else:
```
Comment (criticus) [inline data_ocean/services/koatuu_service.py:119]: `else` is not needed

File: data_ocean/services/koatuu_service.py [hunk 5]
```diff
@@ -15,33 +15,156 @@ class KoatuuConverter(Converter):
...
+                city_koatuu = city_dict[object_district_name]
+                city_koatuu.koatuu = object_koatuu[self.LEVEL_TWO]
+                city_koatuu.save(update_fields=['koatuu'])
+
+    #writing entry to koatuu field in city table
+    def save_to_city_table(self, data, object_koatuu, object_level_number, district_dict, city_dict, object_level_name, district_items_list):
+        for object_name in district_items_list:
+            if not (object_name in district_dict):
+                return
+            else:
```
Comment (criticus) [inline data_ocean/services/koatuu_service.py:141]: not needed

File: data_ocean/services/koatuu_service.py [hunk 6]
```diff
@@ -15,33 +15,156 @@ class KoatuuConverter(Converter):
...
-                region_koatuu = region_koatuu_dict[object_region_name]
-                region_koatuu.koatuu = object_koatuu[self.LEVEL_ONE]
-                region_koatuu.save(update_fields=['koatuu'])
\ No newline at end of file
+    #writing entry to koatuu field in citydistrict table
+    def save_to_citydistrict_table(self, data, object_koatuu, object_level_number, citydistrict_dict, object_level_name, city_items_list, city_dict):
+        for object_name in city_items_list:
+            if not object_name in city_dict:
+                return
+            else:
```
Comment (criticus) [inline data_ocean/services/koatuu_service.py:159]: `else` is not needed after `return`

File: data_ocean/services/koatuu_service.py (hunk 1 above)
Comment (Anntroy) [reply data_ocean/services/koatuu_service.py:68]: It doesn't work without else

File: data_ocean/services/koatuu_service.py (hunk 2 above)
Comment (Anntroy) [reply data_ocean/services/koatuu_service.py:72]: The same

File: data_ocean/services/koatuu_service.py (hunk 3 above)
Comment (Anntroy) [reply data_ocean/services/koatuu_service.py:81]: I put it in a const, and add comment

File: data_ocean/services/koatuu_service.py (hunk 5 above)
Comment (Anntroy) [reply data_ocean/services/koatuu_service.py:141]: I've tried it without else, but it through an error: Unreachable code.

File: data_ocean/services/koatuu_service.py (hunk 4 above)
Comment (Anntroy) [reply data_ocean/services/koatuu_service.py:119]: The same

**Pull request merged**